xbmc/cores/RetroPlayer/buffers/RenderBufferDMA.cpp

```diff
@@ -11,17 +11,18 @@
 #include "ServiceBroker.h"
 #include "utils/BufferObject.h"
 #include "utils/EGLImage.h"
-#include "windowing/gbm/WinSystemGbmEGLContext.h"
+#include "windowing/WinSystem.h"
+#include "windowing/linux/WinSystemEGL.h"
 
-using namespace KODI::WINDOWING::GBM;
 using namespace KODI;
 using namespace RETRO;
 
 CRenderBufferDMA::CRenderBufferDMA(CRenderContext& context, int fourcc)
   : m_context(context),
     m_fourcc(fourcc),
     m_egl(new CEGLImage(
-        static_cast<CWinSystemGbmEGLContext*>(CServiceBroker::GetWinSystem())->GetEGLDisplay())),
+        dynamic_cast<KODI::WINDOWING::LINUX::CWinSystemEGL*>(CServiceBroker::GetWinSystem())
```

Comment:
Please leave `static_cast()`:

* dynamic_cast can return nullptr, which makes f*cking up here a runtime error instead of a programming error, and programming errors can be easier to catch
* dynamic_cast has an extra RTTI hit

Reply:
```
/home/lukas/Documents/git/xbmc/xbmc/cores/RetroPlayer/buffers/RenderBufferDMA.cpp: In constructor ‘KODI::RETRO::CRenderBufferDMA::CRenderBufferDMA(KODI::RETRO::CRenderContext&, int)’:
/home/lukas/Documents/git/xbmc/xbmc/cores/RetroPlayer/buffers/RenderBufferDMA.cpp:24:91: error: invalid static_cast from type ‘CWinSystemBase*’ to type ‘KODI::WINDOWING::LINUX::CWinSystemEGL*’
   24 |         static_cast<KODI::WINDOWING::LINUX::CWinSystemEGL*>(CServiceBroker::GetWinSystem())
      |                                                                                           ^
```

If you want I can break this line up in the constructor so we can check for errors

Reply:
why is static_cast invalid? is CWinSystemEGL not a subclass of CWinSystemBase? maybe it's unhappy with the downcast?

Reply:
>CWinSystemEGL not a subclass of CWinSystemBase

this

Reply:
can you explain how the dynamic_cast works? I haven't had a chance to do a comprehensive read yet

Reply:
`CWinSystemGbmEGLContext` inherits from `CWinSystemEGL` and `CWinSystemBase`
see :
https://github.com/xbmc/xbmc/blob/master/xbmc/windowing/gbm/WinSystemGbmEGLContext.h#L26
and
https://github.com/xbmc/xbmc/blob/master/xbmc/windowing/linux/WinSystemEGL.h#L20